Backend Code/main.py

```diff
@@ -0,0 +1,232 @@
+import sys;
+import math;
+import equivCheck;
+
+class RegularExpressionTreeNode():
+	#Constructor For RegularExpressionTreeNode:
+	def __init__(self, value):
+		self.value = value;
+		self.children = [];
+	
+	def printValue(self):
+		print(self.value);
+
+
+class AndExpressionTreeNode():
+	#Constructor For AndExpressionTreeNode:
+	def __init__(self):
+		self.value = "&";
+		self.children = [];
+
+	def printValue(self):
+		print(self.value);
+
+	def getSatisfyingValues(self):
+		sValue = "";
+		for k in range(0, len(self.children)):
+			currentChild = self.children[k];
+			if(currentChild != None):
+				if(isinstance(currentChild, NotExpressionTreeNode)):
+					sValue += "0";
+				elif(isinstance(currentChild, RegularExpressionTreeNode)):
+					sValue += "1";
+		return sValue;
+
+	def getDistinctVariables(self):
+		return self.children;
+
+
+class OrExpressionTreeNode():
+	#Constructor For OrExpressionTreeNode:
+	def __init__(self):
+		self.value = "|";
+		self.children = [];
+
+	def printValue(self):
+		print(self.value);
+
+	def getSatisfyingValues(self):
+		sValues = [];
+		for k in range(0, len(self.children)):
+			currentChild = self.children[k];
+			if(currentChild != None):
+				cValue = currentChild.getSatisfyingValues();
+				sValues.append(cValue);
+		return sValues;
+
+	def getDistinctVariables(self):

```

Comment:
So why is this function different than the getDistinctVariables() function for the and expression class? @venkatrsrinivas

Reply:
Hello @kd2eom and @hsueha ,
I can definitely add further documentation to the code I have already written. 
Aside from this, the reasons that getDistinctVariables() functions is different for the AndExpression vs. the OrExpression is due to the way the expressions are for our software.
That is, we know that our expressions are in DNF (or I guess CNDF), so that would mean that every AndExpression would only contain distinct variables (i.e., it would never have P & P since that can just be simplified to P). On the other hand, the OrExpression is the "outer" layer if you will of our DNF Expression so to find out how many distinct or total unique variable names (e.g., P, Q, R), I just check to see if the variables from the current AndExpression are already in my allExpressionVariables List Structure. Then I simply return that list after going through all of the AndExpressions embedded within our primary OrExpression. 
I hope that makes more sense, and I will definitely document that with comments soon.
Let me know if you have any further doubts.

Best Regards,
Venkat Srinivas